tinysegmenter/tinysegmenter.py

```diff
@@ -12,181 +12,227 @@
-            score += self._ts(self._BP2, p2 + p3)
-            score += self._ts(self._UW1, w1)
-            score += self._ts(self._UW2, w2)
-            score += self._ts(self._UW3, w3)
-            score += self._ts(self._UW4, w4)
-            score += self._ts(self._UW5, w5)
-            score += self._ts(self._UW6, w6)
-            score += self._ts(self._BW1, w2 + w3)
-            score += self._ts(self._BW2, w3 + w4)
-            score += self._ts(self._BW3, w4 + w5)
-            score += self._ts(self._TW1, w1 + w2 + w3)
-            score += self._ts(self._TW2, w2 + w3 + w4)
-            score += self._ts(self._TW3, w3 + w4 + w5)
-            score += self._ts(self._TW4, w4 + w5 + w6)
-            score += self._ts(self._UC1, c1)
-            score += self._ts(self._UC2, c2)
-            score += self._ts(self._UC3, c3)
-            score += self._ts(self._UC4, c4)
-            score += self._ts(self._UC5, c5)
-            score += self._ts(self._UC6, c6)
-            score += self._ts(self._BC1, c2 + c3)
-            score += self._ts(self._BC2, c3 + c4)
-            score += self._ts(self._BC3, c4 + c5)
-            score += self._ts(self._TC1, c1 + c2 + c3)
-            score += self._ts(self._TC2, c2 + c3 + c4)
-            score += self._ts(self._TC3, c3 + c4 + c5)
-            score += self._ts(self._TC4, c4 + c5 + c6)
-            #  score += self._ts(self._TC5, c4 + c5 + c6)    
-            score += self._ts(self._UQ1, p1 + c1)
-            score += self._ts(self._UQ2, p2 + c2)
-            score += self._ts(self._UQ1, p3 + c3)
-            score += self._ts(self._BQ1, p2 + c2 + c3)
-            score += self._ts(self._BQ2, p2 + c3 + c4)
-            score += self._ts(self._BQ3, p3 + c2 + c3)
-            score += self._ts(self._BQ4, p3 + c3 + c4)
-            score += self._ts(self._TQ1, p2 + c1 + c2 + c3)
-            score += self._ts(self._TQ2, p2 + c2 + c3 + c4)
-            score += self._ts(self._TQ3, p3 + c1 + c2 + c3)
-            score += self._ts(self._TQ4, p3 + c2 + c3 + c4)
-            p = u"O"
-            if score > 0:
-                result.append(word)
-                word = ""
-                p = u"B"
-
-            p1 = p2
-            p2 = p3
-            p3 = p
-            word += seg[i]
-        
-        result.append(word)
-
-        return result
+from __future__ import unicode_literals
+
+import sys
+
+if sys.version_info[0] > 2:
+    unichr = chr
+    unicode = str
+    xrange = range
+
+__all__ = ["tokenize"]
+
+_CHARDICT = {}
+for pat, cat in [
+        (("一", "龠"), 'H'),  # Some chars in this pattern is replaced by next pattern
+        ("一二三四五六七八九十百千万億兆", 'M'),
+        ("々〆ヵヶ", 'H'),
+        (('ぁ','ん'), 'I'),
+        (('ァ','ヴ'), 'K'),
+        ("ーｰ\uff9e", 'K'),
+        (('ｱ', 'ﾝ'), 'K'),
+        (('a', 'z'), 'A'),
+        (('A', 'Z'), 'A'),
+        (('ａ', 'ｚ'), 'A'),
+        (('Ａ', 'Ｚ'), 'A'),
+        (('0', '9'), 'N'),
+        (('０', '９'), 'N')]:
```

Comment:
Could you declare this list as a variable or the constant variable first and then use it ?
It's too long to omit the declaration. By doing so, it would be readble as well.